### PR TITLE
Ready: Adding Edge case tests for CRDTs

### PR DESCRIPTION
**Pull request by alexmoore**
Adding edge case tests for CRDT's listed at https://docs.google.com/a/basho.com/spreadsheets/d/1uQs7hbOkDIamh3APFEDpbdulYs_1A57K2CdETmyugSw/edit#gid=940963132, fixing missing error cases.

### CODE REVIEW

File: src/RiakClient/RiakClient.cs [hunk 1]
```diff
@@ -1653,5 +1675,28 @@ private RiakResult<TResult> UseConnection<TResult>(Func<IRiakConnection, RiakRes
...
+                }
+
+                if (mapUpdate.set_op != null && mapUpdate.set_op.removes != null && mapUpdate.set_op.removes.Count > 0)
+                {
+                    return true;
+                }
+
+                if (mapUpdate.map_op != null)
+                {
+                    AnyNestedRemovalsIn(mapUpdate.map_op.updates);
```
Comment (lukebakken) [inline src/RiakClient/RiakClient.cs]: I think this should be:

```csharp
return AnyNestedRemovalsIn(mapUpdate.map_op.updates);
```

Otherwise the result of recursion won't be bubbled back up.

File: tools/gen-riak-conf [hunk 2]
```diff
@@ -34,6 +34,8 @@ check_crl = off
 
 listener.http.internal = 0.0.0.0:$http_port
 listener.protobuf.internal = 0.0.0.0:$pb_port
+search.solr.start_timeout = 60s
```
Comment (lukebakken) [inline tools/gen-riak-conf]: Good catch on this :smile:

**Comment by lukebakken** [comment]
:+1: once all the buildbots finish

**Comment by alexmoore** [comment]
Everything's ok in build land.
